googleapiclient/http.py

```diff
@@ -1200,9 +1205,10 @@ def _serialize_request(self, request):
     msg = MIMENonMultipart(major, minor)
     headers = request.headers.copy()
 
-    if request.http is not None and hasattr(request.http.request,
-        'credentials'):
-      request.http.request.credentials.apply(headers)
+    if request.http is not None:
+      credentials = _auth.get_credentials_from_http(request.http)
```

Comment:
I guess that's equivalent to checking `_auth.has_access_token` before calling `_auth.apply_credentials`.  It looks like you're already doing that on line 1420, maybe do the same thing here?